templates/.kokoro/build.sh

```diff
@@ -23,11 +23,44 @@ cd ${scriptDir}/..
 # include common functions
 source ${scriptDir}/common.sh
 
+# TODO: Switch monorepo_script_output to main
+# Find the root commit between the two branches
+base_commit=$(git merge-base monorepo_script_output HEAD)
+echo "Base Commit: ${base_commit}"
+
+# Find the files changed from when the PR branched to the last commit
+# Filter for java modules and get all the unique elements
+# grep returns 1 (error code) and exits the pipeline if there is no match
+# If there is no match, it will return true so the rest of the commands can run
+directories=$(git diff --name-only "${base_commit}" HEAD | grep -e 'java-.*' || true)
```

Comment:
`git diff` was used to compare only the changes between latest commit and $base_commit instead of all the files touched between these commits.